src/main/java/org/opensearch/security/compliance/ComplianceIndexingOperationListenerImpl.java

```diff
@@ -90,10 +96,9 @@ public Index preIndex(final ShardId shardId, final Index index) {
             }
 
             if (shard.isReadAllowed()) {
-                try {
-
+                try (ThreadContext.StoredContext ctx = threadPool.getThreadContext().stashContext()) {
+                    threadPool.getThreadContext().putHeader(OPENDISTRO_SECURITY_CONF_REQUEST_HEADER, "true");
```

Comment:
An alternative, more selective solution would be:

```
try (ThreadContext.StoredContext ctx = threadPool.getThreadContext().stashContext()) {
   DocumentAllowList documentAllowList = new DocumentAllowList();
   documentAllowList.add(shardId.getIndexName(), index.id());
   documentAllowList.applyTo(threadPool.getThreadContext();
```

This should already work out of the box, without changes in `DlsFlsFilterLeafReader` or `SecurityFlsDlsIndexSearcherWrapper`.

```